core/src/builder.ts

```diff
@@ -180,7 +185,9 @@ function Builder<Model>({
               if (postBuilder) {
                 transformed = {
                   ...transformed,
-                  ...postBuilder(transformed),
+                  ...postBuilder(transformed, {
```

Comment:
This changes allows for some edge cases with compatibility between new and legacy data models.